batchflow/models/torch/base_mixins.py

```diff
@@ -221,10 +222,255 @@ class OptimalBatchSizeMixin:
-                                   start_batch_size=4, delta_batch_size=4, max_batch_size=128, max_iters=16,
-                                   n=20, frequency=0.05, time_threshold=3, tail_size=20, std_threshold=0.1):
-        """ Compute memory usage for multiple batch sizes. """
+
+    def is_oom_error(self, exception):
+        """ Check whether exception is OOM error """
+        if not (isinstance(exception, RuntimeError) and len(exception.args) == 1):
+            return False
+        return (
+            "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED." in exception.args[0]
+            or "DefaultCPUAllocator: can't allocate memory" in exception.args[0]
+            or ("CUDA" in exception.args[0] and "out of memory" in exception.args[0])
+        )
+
+    def garbage_collection_cuda(self):
+        """ Garbage collection Torch (CUDA) memory. """
+        gc.collect()
+        try:
+            torch.cuda.empty_cache()
+        except RuntimeError as exception:
+            if not self.is_oom_error(exception):
+                raise
+
+
+    def compute_optimal_batch_size(self, method='train', inputs=None, targets=None,
+                                   start_batch_size=4, max_iters=25, factor=2,
+                                   spread=0.2, estimation_method='bruteforce', n=4,
+                                   max_memory=100, pbar='n', tail_size=20,
+                                   frequency=0.05, delta_batch_size=16,
+                                   max_batch_size=1024, max_iters_estimation=2):
+        """ Computes optimal batch size in two steps:
+        1. Calculate batch size estimation using `predictive` or `bruteforce` method.
+        2. Calculate exact optimal batch size using binary search.
+
+        Parameters
+        ----------
+        method: str
+            Defines in which method (`train` or `predict`) the optimal batch size will
+            be computed. Default: `train`.
+
+        estimation_method: str
+            Whether `bruteforce` or `predictive` estimation method will be used.
+
+        inputs: np.array
+            The inputs to the model, that will be used in optimal batch computation.
+            If none, then the placeholder will be created
+
+        targets: np.array
+            The targets to the model, that will be used in optimal batch computation.
+            If none, then we run model.predict() to get targets. Only used in method=`train`
+
+        start_batch_size: int
+            Batch size to start batch_size estimation. If your model is small, you should use larger
+            start_batch_size in order to get accurate optimal batch_size
+
+        max_iters: int
+            Maximum number of binary search iterations.
+
+        factor: int
+            Value by which we multiply batch_size at each iteration. Uses in bruteforce estimation.
+
+        spread: int
+            Defines how wide the interval for binary search will be. Uses in predictive estimation.
```

Comment:
This value is not an int; also, the documentation gives very little insight into how it works / what values should the user pass